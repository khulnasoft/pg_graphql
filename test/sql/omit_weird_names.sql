begin;

    savepoint a;

    create table "@xyz"( id int primary key);

    select jsonb_pretty(
        graphql.resolve($$
            {
              __schema {
                types {
                  name
                }
              }
            }
        $$)
    );

    rollback to savepoint a;

    create table xyz( "! q" int primary key);
    select jsonb_pretty(
        graphql.resolve($$
            {
              __type(name: "Xyz") {
                fields {
                  name
                }
              }
            }
        $$)
    );

    rollback to savepoint a;

rollback;
