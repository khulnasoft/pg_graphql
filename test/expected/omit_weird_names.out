begin;
    savepoint a;
    create table "@xyz"( id int primary key);
    select jsonb_pretty(
        graphql.resolve($$
            {
              __schema {
                types {
                  name
                }
              }
            }
        $$)
    );
                   jsonb_pretty                    
---------------------------------------------------
 {                                                +
     "data": {                                    +
         "__schema": {                            +
             "types": [                           +
                 {                                +
                     "name": "BigFloat"           +
                 },                               +
                 {                                +
                     "name": "BigFloatFilter"     +
                 },                               +
                 {                                +
                     "name": "BigFloatListFilter" +
                 },                               +
                 {                                +
                     "name": "BigInt"             +
                 },                               +
                 {                                +
                     "name": "BigIntFilter"       +
                 },                               +
                 {                                +
                     "name": "BigIntListFilter"   +
                 },                               +
                 {                                +
                     "name": "Boolean"            +
                 },                               +
                 {                                +
                     "name": "BooleanFilter"      +
                 },                               +
                 {                                +
                     "name": "BooleanListFilter"  +
                 },                               +
                 {                                +
                     "name": "Cursor"             +
                 },                               +
                 {                                +
                     "name": "Date"               +
                 },                               +
                 {                                +
                     "name": "DateFilter"         +
                 },                               +
                 {                                +
                     "name": "DateListFilter"     +
                 },                               +
                 {                                +
                     "name": "Datetime"           +
                 },                               +
                 {                                +
                     "name": "DatetimeFilter"     +
                 },                               +
                 {                                +
                     "name": "DatetimeListFilter" +
                 },                               +
                 {                                +
                     "name": "FilterIs"           +
                 },                               +
                 {                                +
                     "name": "Float"              +
                 },                               +
                 {                                +
                     "name": "FloatFilter"        +
                 },                               +
                 {                                +
                     "name": "FloatListFilter"    +
                 },                               +
                 {                                +
                     "name": "ID"                 +
                 },                               +
                 {                                +
                     "name": "IDFilter"           +
                 },                               +
                 {                                +
                     "name": "Int"                +
                 },                               +
                 {                                +
                     "name": "IntFilter"          +
                 },                               +
                 {                                +
                     "name": "IntListFilter"      +
                 },                               +
                 {                                +
                     "name": "JSON"               +
                 },                               +
                 {                                +
                     "name": "Node"               +
                 },                               +
                 {                                +
                     "name": "Opaque"             +
                 },                               +
                 {                                +
                     "name": "OpaqueFilter"       +
                 },                               +
                 {                                +
                     "name": "OrderByDirection"   +
                 },                               +
                 {                                +
                     "name": "PageInfo"           +
                 },                               +
                 {                                +
                     "name": "Query"              +
                 },                               +
                 {                                +
                     "name": "String"             +
                 },                               +
                 {                                +
                     "name": "StringFilter"       +
                 },                               +
                 {                                +
                     "name": "StringListFilter"   +
                 },                               +
                 {                                +
                     "name": "Time"               +
                 },                               +
                 {                                +
                     "name": "TimeFilter"         +
                 },                               +
                 {                                +
                     "name": "TimeListFilter"     +
                 },                               +
                 {                                +
                     "name": "UUID"               +
                 },                               +
                 {                                +
                     "name": "UUIDFilter"         +
                 },                               +
                 {                                +
                     "name": "UUIDListFilter"     +
                 },                               +
                 {                                +
                     "name": "__Directive"        +
                 },                               +
                 {                                +
                     "name": "__DirectiveLocation"+
                 },                               +
                 {                                +
                     "name": "__EnumValue"        +
                 },                               +
                 {                                +
                     "name": "__Field"            +
                 },                               +
                 {                                +
                     "name": "__InputValue"       +
                 },                               +
                 {                                +
                     "name": "__Schema"           +
                 },                               +
                 {                                +
                     "name": "__Type"             +
                 },                               +
                 {                                +
                     "name": "__TypeKind"         +
                 }                                +
             ]                                    +
         }                                        +
     }                                            +
 }
(1 row)

    rollback to savepoint a;
    create table xyz( "! q" int primary key);
    select jsonb_pretty(
        graphql.resolve($$
            {
              __type(name: "Xyz") {
                fields {
                  name
                }
              }
            }
        $$)
    );
             jsonb_pretty             
--------------------------------------
 {                                   +
     "data": {                       +
         "__type": {                 +
             "fields": [             +
                 {                   +
                     "name": "nodeId"+
                 }                   +
             ]                       +
         }                           +
     }                               +
 }
(1 row)

    rollback to savepoint a;
rollback;
